File: Lecture03_Layouts/AfterWorkshop_Java/app/src/main/java/com/lavima/afterworkshop_java/MainActivity.java
package com.lavima.afterworkshop_java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button buttonLong = findViewById(R.id.buttonLong);
        Button buttonMulti = findViewById(R.id.buttonMulti);
        Button buttonFrame = findViewById(R.id.buttonFrame);

        buttonLong.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(view.getContext(), ScrollLongActivity.class));
            }
        });
        buttonMulti.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(view.getContext(), ScrollMultiActivity.class));
            }
        });
        buttonFrame.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(view.getContext(), FrameActivity.class));
            }
        });
    }
}

File: Lecture02_Intents/AfterLecture_Kotlin/app/src/main/java/no/hiof/larseknu/playingwithintents/MainActivity.kt
package no.hiof.larseknu.playingwithintents

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    companion object {
        const val KEY_NAME = "name"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        startOtherActivityButton.setOnClickListener { view ->
            val intent = Intent(view.context, OtherActivity::class.java)

            val name = nameEditText.text.toString()

            intent.putExtra(KEY_NAME, name)

            startActivity(intent)
        }
    }

    fun openWebPage(view: View) {
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse("http://hiof.no"))

        startActivity(intent)
    }

    fun openMaps(view: View) {
        //Uri uri = Uri.parse("https://maps.google.com?q=59.128708,11.353176");

        val uri = Uri.parse("geo:59.128708,11.353176")

        val intent = Intent(Intent.ACTION_VIEW, uri)

        startActivity(intent)
    }
}

